File: JavaScript/7js 1/dropCap.js
/*
|--------------------------------------------------------------------------
| Dropcaps
|--------------------------------------------------------------------------
|
*/

function dropCap(n) {
   let arr = n.split(" ");
   for (let i = 0; i < arr.length; i++) {
      if (arr[i].length > 2) {
         arr[i] = arr[i].slice(0, 1).toUpperCase() + arr[i].slice(1).toLowerCase();
      }
   }
   return arr.join(" ");
}


function dropCap(n) {
   return n.replace(/\b(\w)(\w{2,})/g, (_, $1, $2) => $1.toUpperCase() + $2.toLowerCase());
}


function dropCap(str) {
   return str
      .split(" ")
      .map(function (w) {
         return w.length > 2 ? w[0].toUpperCase() + w.slice(1).toLowerCase() : w;
      })
      .join(" ");
}


console.log(dropCap("Apple Banana"), "Apple Banana");
console.log(dropCap("Apple"), "Apple");
console.log(dropCap(""), "");
console.log(dropCap("of"), "of");
console.log(dropCap("  leading spaces"), "  Leading Spaces");
console.log(dropCap("trailing spaces   "), "Trailing Spaces   ");
console.log(dropCap("ALL CAPS CRAZINESS"), "All Caps Craziness");
console.log(dropCap("rAnDoM CaPs CrAzInEsS"), "Random Caps Craziness");

File: Python/7py 1/drop_cap.py
"""
|--------------------------------------------------------------------------
| Dropcaps
|--------------------------------------------------------------------------
|
"""

def drop_cap(n):
   arr = n.split(" ")
   for i in range(len(arr)):
      if len(arr[i]) > 2:
         arr[i] = arr[i][0].upper() + arr[i][1:].lower()
   return " ".join(arr)


def drop_cap(str_):
   return ' '.join( w.capitalize() if len(w) > 2 else w for w in str_.split(' ') )


from re import sub
def drop_cap(s):
   return sub(r'\S{3,}', lambda m: m.group(0).title(), s)


print(drop_cap("Apple Banana"), "Apple Banana")
print(drop_cap("Apple"), "Apple")
print(drop_cap(""), "")
print(drop_cap("of"), "of")
print(drop_cap("  leading spaces"), "  Leading Spaces")
print(drop_cap("trailing spaces   "), "Trailing Spaces   ")
print(drop_cap("ALL CAPS CRAZINESS"), "All Caps Craziness")
print(drop_cap("rAnDoM CaPs CrAzInEsS"), "Random Caps Craziness")